clientv3/watch.go

```diff
@@ -612,6 +601,20 @@ func (w *watchGrpcStream) serveSubstream(ws *watcherStream, resumec chan struct{
 				// shutdown from closeSubstream
 				return
 			}
+
+			if wr.Created {
+				select {
+				case ws.initReq.retc <- ws.outc:
+				default:
+				}
+
+				// send creation event and only if requested
+				if ws.initReq.createdNotify {
```

Comment:
There should be code to check if `retc <- ws.outc` went through this time before sending this message. Otherwise, this still sends a create notification every time the watcher resumes from a disconnect (and potentially block forever since `ws.outc` isn't buffered to do that).